Description: Build the X.Org driver only for the selected system X Server version.
Author: Michael Meskes <meskes@debian.org>, Felix Geyer <debfx-pkg@fobos.de>

diff -urNad virtualbox-ose-3.1.0-dfsg~/src/VBox/Additions/common/crOpenGL/Makefile.kmk virtualbox-ose-3.1.0-dfsg/src/VBox/Additions/common/crOpenGL/Makefile.kmk
--- virtualbox-ose-3.1.0-dfsg~/src/VBox/Additions/common/crOpenGL/Makefile.kmk	2009-10-07 11:27:13.000000000 +0200
+++ virtualbox-ose-3.1.0-dfsg/src/VBox/Additions/common/crOpenGL/Makefile.kmk	2009-10-07 13:44:19.145463290 +0200
@@ -63,16 +63,12 @@
 VBoxOGL_INCS           = .
 if1of ($(KBUILD_TARGET), linux solaris freebsd)
  VBoxOGL_INCS     += \
-	$(VBOX_PATH_X11_ROOT)/libXdamage-1.1 \
-	$(VBOX_PATH_X11_ROOT)/libXcomposite-0.4.0 \
-	$(VBOX_PATH_X11_ROOT)/libXfixes-4.0.3 \
-	$(VBOX_PATH_X11_ROOT)/damageproto-1.1.0 \
-	$(VBOX_PATH_X11_ROOT)/compositeproto-0.4 \
-	$(VBOX_PATH_X11_ROOT)/fixesproto-4.0 \
-	$(VBOX_PATH_X11_ROOT)/libx11-1.1.5-other \
-	$(VBOX_PATH_X11_ROOT)/1.3/xorg \
+	/usr/include/x11 \
+	/usr/include/xorg \
+	/usr/include/pixman-1 \
 	$(VBOX_MESA_INCS) \
-	$(PATH_ROOT)/src/VBox/Additions/x11/x11include/libdrm-2.4.13
+	/usr/include/drm \
+	/usr/include/libdrm
  VBoxOGL_DEFS     += VBOX_NO_NATIVEGL
 endif
 
--- virtualbox-ose-3.1.51-dfsg.orig/src/VBox/Additions/common/VBoxGuestLib/Makefile.kmk
+++ virtualbox-ose-3.1.51-dfsg/src/VBox/Additions/common/VBoxGuestLib/Makefile.kmk
@@ -40,8 +40,8 @@
 	VBoxGuestR3LibShared
 ifndef VBOX_ONLY_TESTSUITE
  if1of ($(KBUILD_TARGET), freebsd linux netbsd openbsd)
-  LIBRARIES += \
-  	VBoxGuestR3LibXFree86
+#  LIBRARIES += \
+#  	VBoxGuestR3LibXFree86
  endif
 endif
 
diff -Nur virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/Makefile.kmk virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/Makefile.kmk
--- virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/Makefile.kmk	2010-02-13 15:19:45.000000000 +0100
+++ virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/Makefile.kmk	2010-03-01 00:13:12.324136362 +0100
@@ -17,6 +17,10 @@
 SUB_DEPTH = ../../../..
 include $(KBUILD_PATH)/subheader.kmk
 
+ifn1of ($(XSERVER_VERSION), 70 71 13 14 15 16 17 18)
+ XSERVER_VERSION := 17
+endif
+
 # Include sub-makefiles.
 if1of ($(KBUILD_TARGET), freebsd linux netbsd openbsd solaris)
  include $(PATH_SUB_CURRENT)/VBoxClient/Makefile.kmk
diff -Nur virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxmouse/Makefile.kmk virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxmouse/Makefile.kmk
--- virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxmouse/Makefile.kmk	2010-02-13 15:20:20.000000000 +0100
+++ virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxmouse/Makefile.kmk	2010-03-01 00:05:56.464134113 +0100
@@ -23,7 +23,6 @@
 # vboxmouse_drv
 #
 if1of ($(KBUILD_TARGET), linux)
- SYSMODS += vboxmouse_drv
  vboxmouse_drv_TEMPLATE = VBOXGUESTR3XF86MOD
  vboxmouse_drv_DEFS.linux = linux
  vboxmouse_drv_DEFS.x86 += __i386__
@@ -59,7 +58,6 @@
 #
 # vboxmouse_drv_70
 #
-DLLS += vboxmouse_drv_70
 vboxmouse_drv_70_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_70_DEFS = \
 	XFree86Server IN_MODULE XFree86Module XFree86LOADER XINPUT XORG_7X
@@ -80,7 +78,6 @@
 #
 # vboxmouse_drv_71
 #
-DLLS += vboxmouse_drv_71
 vboxmouse_drv_71_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_71_DEFS := $(vboxmouse_drv_70_DEFS)
 vboxmouse_drv_71_INCS := \
@@ -97,7 +94,6 @@
 #
 # vboxmouse_drv_13
 #
-DLLS += vboxmouse_drv_13
 vboxmouse_drv_13_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_13_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_13_INCS := \
@@ -112,7 +108,6 @@
 #
 # vboxmouse_drv_14
 #
-DLLS += vboxmouse_drv_14
 vboxmouse_drv_14_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_14_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_14_INCS := \
@@ -133,7 +128,6 @@
 #
 # vboxmouse_drv_15
 #
-DLLS += vboxmouse_drv_15
 vboxmouse_drv_15_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_15_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_15_INCS := \
@@ -148,7 +142,6 @@
 #
 # vboxmouse_drv_16
 #
-DLLS += vboxmouse_drv_16
 vboxmouse_drv_16_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_16_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_16_INCS := \
@@ -164,7 +157,6 @@
 #
 # vboxmouse_drv_17
 #
-DLLS += vboxmouse_drv_17
 vboxmouse_drv_17_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_17_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 ## @todo replace $(VBOX_PATH_X11_ROOT)/xorg-server-1.6.0-local
@@ -184,7 +176,6 @@
 #
 # vboxmouse_drv_18
 #
-DLLS += vboxmouse_drv_18
 vboxmouse_drv_18_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_18_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 ## @todo replace $(VBOX_PATH_X11_ROOT)/xorg-server-1.6.0-local
@@ -201,6 +192,14 @@
 	vboxmouse_15.c
 
 
+vboxmouse_drv_$(XSERVER_VERSION)_NAME := vboxmouse_drv
+vboxmouse_drv_$(XSERVER_VERSION)_INCS := \
+	/usr/include/x11 \
+	/usr/include/xorg \
+	/usr/include/pixman-1
+DLLS += vboxmouse_drv_$(XSERVER_VERSION)
+
+
 # Check the undefined symbols in the X.Org modules against lists of allowed
 # symbols.  Not very elegant, but it will catch problems early.
 ifdef VBOX_WITH_TESTCASES
diff -Nur virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxvideo/Makefile.kmk virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxvideo/Makefile.kmk
--- virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2010-02-13 15:20:20.000000000 +0100
+++ virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2010-03-01 00:07:13.334156070 +0100
@@ -22,7 +22,6 @@
 # vboxvideo_drv
 #
 if1of ($(KBUILD_TARGET), linux)
- SYSMODS += vboxvideo_drv
  vboxvideo_drv_TEMPLATE = VBOXGUESTR3XF86MOD
  vboxvideo_drv_DEFS.linux = linux
  vboxvideo_drv_DEFS.x86 = __i386__
@@ -83,7 +82,6 @@
 #         base keywords instead of using .solaris or .linux.
 #         Also it is *important* to use := and not = when deriving a properity.
 #
-DLLS += vboxvideo_drv_70
 vboxvideo_drv_70_TEMPLATE = VBOXGUESTR3XORGMOD
 if1of ($(KBUILD_TARGET), linux)
  vboxvideo_drv_70_CFLAGS += \
@@ -106,7 +104,6 @@
 #
 # vboxvideo_drv_71
 #
-DLLS += vboxvideo_drv_71
 vboxvideo_drv_71_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_71_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_71_DEFS := $(vboxvideo_drv_70_DEFS)
@@ -122,7 +119,6 @@
 #
 # vboxvideo_drv_13
 #
-DLLS += vboxvideo_drv_13
 vboxvideo_drv_13_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_13_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_13_DEFS := $(vboxvideo_drv_70_DEFS) VBOXVIDEO_13
@@ -145,7 +141,6 @@
 # This uses the same code (vboxvideo_13.c) as 1.3, but is built
 # for 1.4 as well in case there should be any relevant header changes.
 #
-DLLS += vboxvideo_drv_14
 vboxvideo_drv_14_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_14_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_14_DEFS := $(vboxvideo_drv_13_DEFS)
@@ -165,7 +160,6 @@
 #
 # vboxvideo_drv_15
 #
-DLLS += vboxvideo_drv_15
 vboxvideo_drv_15_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_15_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_15_DEFS := $(vboxvideo_drv_13_DEFS) NO_ANSIC PCIACCESS \
@@ -193,7 +187,6 @@
 #
 # vboxvideo_drv_16
 #
-DLLS += vboxvideo_drv_16
 vboxvideo_drv_16_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_16_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_16_DEFS := $(vboxvideo_drv_15_DEFS)
@@ -212,7 +205,6 @@
 #
 # vboxvideo_drv_17
 #
-DLLS += vboxvideo_drv_17
 vboxvideo_drv_17_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_17_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_17_DEFS := $(vboxvideo_drv_15_DEFS)
@@ -238,7 +230,6 @@
 #
 # vboxvideo_drv_18
 #
-DLLS += vboxvideo_drv_18
 vboxvideo_drv_18_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_18_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_18_DEFS := $(vboxvideo_drv_15_DEFS)
@@ -261,6 +252,22 @@
 vboxvideo_drv_18_SOURCES := $(vboxvideo_drv_15_SOURCES)
 
 
+vboxvideo_drv_$(XSERVER_VERSION)_NAME := vboxvideo_drv
+vboxvideo_drv_$(XSERVER_VERSION)_INCS := \
+	/usr/include/xorg \
+	/usr/include/x11 \
+	/usr/include/pixman-1 \
+	/usr/include/X11/dri \
+	/usr/include/drm \
+	/usr/include/libdrm
+DLLS += vboxvideo_drv_$(XSERVER_VERSION)
+
+# required for lenny backports
+ifeq ($(XSERVER_VERSION),14)
+	vboxvideo_drv_$(XSERVER_VERSION)_INCS += $(VBOX_PATH_X11_XORG_1_4)/xorg
+endif
+
+
 # Check the undefined symbols in the X.Org modules against lists of allowed
 # symbols.  Not very elegant, but it will catch problems early.
 ifdef VBOX_WITH_TESTCASES
